### PR TITLE
fix: fix reject flow

### PR DESCRIPTION
**Pull request by chmyga**
* Add reject flow to connectors list

* Add schemas for all connectors, which need this

* Fix generic in javajet

### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant

### CODE REVIEW

File: component-studio-integration/src/main/java/org/talend/sdk/component/studio/ComponentModel.java [hunk 1]
```diff
@@ -238,7 +238,7 @@ public String getVersion() {
     @Override // TODO This is dummy implementation. Correct impl should be added soon
     public List<? extends IElementParameter> createElementParameters(final INode node) {
         ElementParameterCreator creator = new ElementParameterCreator(this, detail, node);
-        List<IElementParameter> parameters = (List<IElementParameter>) creator.createParameters();
+        List<IElementParameter> parameters = (List<IElementParameter>) creator.createParameters(node);
```
Comment (gonchar-ivan) [inline component-studio-integration/src/main/java/org/talend/sdk/component/studio/ComponentModel.java]: seems, node parameter is not required here. It is passed to ElementParameterCreator constructor.
Thus, you don't need to pass it to ElementParameterCreator.addComponentParameters() method. You may use the field directly

File: component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java [hunk 2]
```diff
@@ -100,12 +101,36 @@ private void addComponentParameters() {
...
+    }
 
+    /**
+     * Check whether all required schema settings are created and create ones we still need depending on connectors we
+     * have.
+     * 
+     * @param node
+     * @param mainSettingsCreator
+     */
+    protected void checkSchemaProperties(final INode node, final SettingsCreator mainSettingsCreator) {
```
Comment (gonchar-ivan) [inline component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java]: seems, `node` parameter is not required here. It is already available as a class field

File: component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java [hunk 3]
```diff
@@ -100,12 +101,36 @@ private void addComponentParameters() {
...
+        // Get all schema parameters created for current component
+        final Set<String> schemasPresent = parameters
+                .stream()
+                .filter(p -> EParameterFieldType.SCHEMA_TYPE.equals(p.getFieldType()))
+                .map(p -> p.getContext())
```
Comment (gonchar-ivan) [inline component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java]: seems, you may use method reference here:
```
IElementParameter::getContext()
```

File: component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java [hunk 4]
```diff
@@ -100,12 +101,36 @@ private void addComponentParameters() {
...
+                .collect(Collectors.toSet());
+        // Get all connectors without schema parameter for them
+        final Set<String> connectorNames = component
+                .createConnectors(node)
+                .stream()
+                .filter(c -> c.getDefaultConnectionType().hasConnectionCategory(IConnectionCategory.FLOW)
+                        && !schemasPresent.contains(c.getName()))
+                .map(c -> c.getName())
```
Comment (gonchar-ivan) [inline component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java]: method reference may be used here

File: component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java [hunk 5]
```diff
@@ -100,12 +101,36 @@ private void addComponentParameters() {
...
+                .collect(Collectors.toSet());
+        // Create schema parameter for each connector without schema parameter
+        for (String connectorWithoutSchema : connectorNames) {
+            parameters.add(mainSettingsCreator.createSchemaParameter(connectorWithoutSchema,
```
Comment (gonchar-ivan) [inline component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java]: just a note:
This loop may also be rewritten in following way.
```
connectorNames.forEach(c -> parameters.add(mainSettingsCreator.createSchemaParameter(c, "SCHEMA_" + c)));
```
However, your current code looks clearer for me than a code with "forEach".

File: component-studio-integration/src/main/java/org/talend/sdk/component/studio/ComponentModel.java (hunk 1 above)
Comment (chmyga) [reply component-studio-integration/src/main/java/org/talend/sdk/component/studio/ComponentModel.java]: That's reasonable! thanks! I missed that. I'll fix it.

File: component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java (hunk 5 above)
Comment (chmyga) [reply component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java]: I would leave that as is, if you don't mind.

File: component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java (hunk 4 above)
Comment (chmyga) [reply component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java]: fixed.

File: component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java (hunk 3 above)
Comment (chmyga) [reply component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java]: fixed

File: component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java (hunk 2 above)
Comment (chmyga) [reply component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java]: fixed

**Comment by chmyga** [comment]
Do not merge yet, please. Found a problem with this PR.

**Pull request merged**